rsrcdump/resconverters.py

```diff
@@ -39,70 +45,94 @@ def parse_record(self, data, offset):
         return str_list
 
 class TextConverter(ResourceConverter):
-    def convert(self, res, res_map):
+    __slots__: Tuple = tuple()
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> str:
         return res.data.decode("macroman")
 
 class IcnsConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.icns')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         return res.data
 
 class SoundToAiffConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.aiff')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         return convert_snd_to_aiff(res.data, res.name)
 
 class PictConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.png')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         w, h, data = convert_pict_to_image(res.data)
         return pack_png(data, w, h)
 
 class CicnConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.png')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         w, h, data = convert_cicn_to_image(res.data)
         return pack_png(data, w, h)
 
 class PpatConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.png')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         w, h, data = convert_ppat_to_image(res.data)
         return pack_png(data, w, h)
 
 class SicnConverter(ResourceConverter):
-    def __init__(self):
+    __slots__: Tuple = tuple()
+    def __init__(self) -> None:
         super().__init__(separate_file='.png')
 
-    def convert(self, res, res_map):
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> bytes:
         w, h, data = convert_sicn_to_image(res.data)
         return pack_png(data, w, h)
 
 class TemplateConverter(ResourceConverter):
-    def convert(self, res, res_map):
+    __slots__: Tuple = tuple()
+    def convert(self, res: Resource,
+                res_map: Dict[bytes, Dict[int, Resource]]) -> List[Dict[str, Union[str, bytes]]]:
```

Comment:
unpack_pstr is guaranteed to return a str in this case
```suggestion
                res_map: dict[bytes, dict[int, Resource]]) -> list[dict[str, str]]:
```